space.crankshaft.Crankshaft.yml

```diff
@@ -0,0 +1,69 @@
+app-id: space.crankshaft.Crankshaft
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+  - org.freedesktop.Sdk.Extension.node18
+
+command: crankshaft
+
+finish-args:
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --talk-name=org.freedesktop.Flatpak
+  # Make $XDG_STATE_HOME work on Flatpak 1.12 and lower
+  - --persist=.local/state
+  - --unset-env=XDG_STATE_HOME
+  # Non-Flatpak Steam
+  - --filesystem=xdg-data/Steam:rw
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+
+  - name: js-beautify
+    buildsystem: simple
+    build-options:
+      append-path: "/usr/lib/sdk/node18/bin"
+      env:
+        npm_config_nodedir: /usr/lib/sdk/node18
+    build-commands:
+      - npm ci --offline --cache=$FLATPAK_BUILDER_BUILDDIR/flatpak-node/npm-cache
+      - make js
+      - install -Dm00755 /usr/lib/sdk/node18/bin/node $FLATPAK_DEST/bin/node
+      - install -Dm00755 js-beautify $FLATPAK_DEST/bin/js-beautify
+    sources:
+      - js-beautify-sources.json
+      - type: git
+        url: https://github.com/coolavery/js-beautify.git
+        commit: bc4dd9e787ff8ee9f4709d76633d71e44c4ac4e2
```

Comment:
Sorry, just realized now that this only has a commit, no tag. Does the tag + commit requirement also apply to dependencies, or only the main application? I can add a tag if needed.

Reply:
Just commit is fine. It's important it's reproducible, otherwise it may fail to build in a week because of some changes.